ThirdPartyAdapters/chartboost/chartboost/src/main/java/com/google/ads/mediation/chartboost/ChartboostSingleton.java

```diff
@@ -208,27 +265,44 @@ private static void startChartboost(Activity activity, ChartboostParams params,
 
         if (!mIsChartboostInitialized) {
             mIsChartboostInitializing = true;
-            Chartboost.startWithAppId(activity, params.getAppId(), params.getAppSignature());
-
-            Chartboost.setMediation(Chartboost.CBMediation.CBMediationAdMob,
-                    BuildConfig.VERSION_NAME);
-            Chartboost.setLoggingLevel(CBLogging.Level.INTEGRATION);
+            Chartboost.startWithAppId(context, params.getAppId(), params.getAppSignature());
+            Chartboost.setMediation(Chartboost.CBMediation.CBMediationAdMob, BuildConfig.SDK_VERSION, "8.0.1.0");
+            Chartboost.setLoggingLevel(CBLogging.Level.ALL);
             Chartboost.setDelegate(getInstance());
             Chartboost.setAutoCacheAds(false);
-
-            // Chartboost depends on Activity's lifecycle events to initialize its SDK. Chartboost
-            // requires onCreate, onStart and onResume callbacks to initialize its SDK.  By the time
-            // AdMob SDK requests this adapter to initialize the SDK, all three callbacks might have
-            // already been called. So, we call Chartboost's onCreate, onStart and onResume methods
-            // so that the Chartboost SDK will be initialized.
-            Chartboost.onCreate(activity);
-            Chartboost.onStart(activity);
-            Chartboost.onResume(activity);
+            initBanner(context, params, bannerListener);
+            adapterDelegate.didInitialize();
         } else {
+            initBanner(context, params, bannerListener);
             adapterDelegate.didInitialize();
         }
+
     }
 
+    /**
+     * Initialize {@link ChartboostBanner}. Can only be done after Chartboost sdk is initialized
+     * and before didInitialize
+     * @param context
+     * @param params
+     * @param bannerListener
+     */
+    private static void initBanner(Context context, ChartboostParams params, ChartboostBannerListener bannerListener) {
+        if(bannerListener != null) {
+            String location = params.getLocation();
+
+            //Attach object to layout
+            LinearLayout bannerContainer = new LinearLayout(context);
+            LinearLayout.LayoutParams paramsLayout = new LinearLayout.LayoutParams(
+                    LinearLayout.LayoutParams.WRAP_CONTENT,
+                    LinearLayout.LayoutParams.WRAP_CONTENT);
+            paramsLayout.gravity = Gravity.CENTER_HORIZONTAL;
+
+            ChartboostBanner chartboostBanner = new ChartboostBanner(context, location, BannerSize.STANDARD, bannerListener);
+            chartboostBanner.setAutomaticallyRefreshesContent(true);
```

Comment:
This probably should be set to false. Btw true is the default right?

Reply:
yes, true is default